commons-packet/commons-packet-service/src/main/java/io/mosip/commons/packetmanager/config/HazelcastHealthIndicator.java

```diff
@@ -0,0 +1,45 @@
+package io.mosip.commons.packetmanager.config;
+
+import java.lang.reflect.Method;
+
+import com.hazelcast.core.HazelcastInstance;
+
+import org.springframework.boot.actuate.health.AbstractHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import org.springframework.util.ReflectionUtils;
+
+@Component
+@ConditionalOnProperty(value = "packetmanager.hazelcast.disable.health.check", matchIfMissing = true)
+public class HazelcastHealthIndicator extends AbstractHealthIndicator {
+
+    private final HazelcastInstance hazelcast;
+
+    public HazelcastHealthIndicator(HazelcastInstance hazelcast) {
+        super("Hazelcast health check failed");
+        this.hazelcast = hazelcast;
+    }
+
+    @Override
+    protected void doHealthCheck(Health.Builder builder) {
+        this.hazelcast.executeTransaction((context) -> {
+            builder.up().withDetail("name", this.hazelcast.getName()).withDetail("uuid", extractUuid())
+            .withDetail("members", hazelcast.getCluster().getMembers().size());
+            return null;
+        });
+    }
+
+    private String extractUuid() {
+        try {
+            return this.hazelcast.getLocalEndpoint().getUuid();
+        }
+        catch (NoSuchMethodError ex) {
+            Method endpointAccessor = ReflectionUtils.findMethod(HazelcastInstance.class, "getLocalEndpoint");
```

Comment:
If method not found can't we simply log and return null, why should we use ReflectionUtils to find and call? Also we if we doing this, then we should check for nageitive caseslike method not found, arguments differnt etc for both getLocalEnpoint and getUuid

Reply:
This code is taken from hazelcast documentation so used as is without change.